### ISSUE TITLE
zip + unique doesn't work

### ISSUE THREAD

**Issue by jaredgrubb** [ericniebler/range-v3#561]
I'm trying to use unique on a zip view and it says it's not erasable.

    std::vector<std::string> keys;
    std::vector<int> values;

    ranges::view::zip(keys, values)
    | ranges::action::sort;
    | ranges::action::unique
    ;

I think that zip is calling itself an Input range and not necessarily a Forward range if its underlying ranges are forward?

**Comment by ericniebler** [comment]
No, zip is random-access if its underlying ranges are random-access. `action::unique` is expecting a container from which it can erase elements. The result of `view::zip` is not that. I _think_ it could be. It's an interesting suggestion at any rate.

**Comment by Victor-Savu** [comment]
Sorry if I this is a stupid question, but couldn't unique be a view? Since it expects the input range to be sorted, it only needs O(1) state to skip over equivalent elements.

**Comment by tivek** [comment]
There is a `unique::view`. In fact your code should work with this minimal change:
```
ranges::view::zip(keys, values) | ranges::action::sort | ranges::view::unique;
```
